### ISSUE TITLE
player:SendListInventory(creature, VENDOR_ID1)  doesn't show vendor panel

### ISSUE THREAD

**Issue by rtippenhauer** [New-HavenWotLK/AzerothCore-lua-MultiVendor#1]
I've created the NPC, updated the script as follows and when I select one of the entries on the list, nothing happens, not even the close menu seems to work.  I'm running AzerothCore in docker, I've had it running about 2 weeks.  Both AzerothCore and Module Eluna are up to date with 'origin/master'.  I've tried adding broadcast statements to see that the choice I'm selecting is correct, but it looks like after the player:SendListInventory(creature, VENDOR_IDx) the script crashes.  I've just starting using eluna, so I'm not very good at troubleshooting and the only other example of SendListInventory I see just has (creature) which I'm assuming will have the NPC show it's inventory panel.

--[[Script done by Psychonautek(Krisande#5411 @ Discord)]]-- 

local EVENT_ONGOSSIPHELLO       = 1

local EVENT_ONGOSSIPSELECT      = 2

local NPC_MULTIVENDOR           = 200000

local VENDOR_ID1                = 34785 -- insert your vendor id from npc_vendor table here!

local VENDOR_ID2                = 30716-- insert another vendor id from npc_vendor table here! 

local VENDOR_ID3 	        = 40607

local VENDOR_ID4 	        = 3059

-- in case you want to add more vendors just add them via local VENDOR_ID3, local VENDOR_ID4 and so on =)

local function MultiVendorOnGossipHello(event, player, creature)

    player:SendListInventory(creature, VENDOR_ID3) 

    player:GossipClearMenu()

    player:GossipMenuAddItem(1, "Cooking Trainer", 1, 1)

    player:GossipMenuAddItem(1, "Inscription Trainer", 1, 2)

    player:GossipMenuAddItem(1, "Trade Goods kawnie", 1, 3)

    player:GossipMenuAddItem(1, "Warrior Trainer", 1, 4)

    -- to add more gossip items just copy a line up above and change the last value in there... so if you copy the line with 2 at the end... make the 2 a 3 and so on =)

    player:GossipMenuAddItem(1, "Close", 1, 100)

    player:GossipSendMenu(1, creature, 1)

end

local function MultiVendorOnGossipSelect(event, player, creature, sender, intid, code, menu_id)    

    if (intid == 1) then

        player:SendListInventory(creature, VENDOR_ID1) 

    elseif (intid == 2) then

        player:SendListInventory(creature, VENDOR_ID2) 

    elseif (intid == 3) then

        player:SendListInventory(creature, VENDOR_ID3) 

    elseif (intid == 4) then

        player:SendListInventory(creature, VENDOR_ID4)

        -- to add more selection options just copy the line up above, change the intid according to the last number value at the GossipMenuAddItem and change the vendorid =)

    

    elseif (intid == 100) then

        player:GossipComplete()

    end

end

RegisterCreatureGossipEvent(NPC_MULTIVENDOR, EVENT_ONGOSSIPHELLO, MultiVendorOnGossipHello)

RegisterCreatureGossipEvent(NPC_MULTIVENDOR, EVENT_ONGOSSIPSELECT, MultiVendorOnGossipSelect)

**Comment by Ryn595** [comment]
I was able to get it working by changing the npcflag value for the multivendor in creature_template from 
1 - the flag for Gossip 
to 
129 - the flag for Vendor(128) + Gossip(1) 

This would be the new default MultiVendor.SQL with just the npcflag value changed:
```
INSERT INTO `creature_template`(`entry`, `difficulty_entry_1`, `difficulty_entry_2`, `difficulty_entry_3`, `KillCredit1`, `KillCredit2`, `modelid1`, `modelid2`, `modelid3`, `modelid4`, `name`, `subname`, `IconName`, `gossip_menu_id`, `minlevel`, `maxlevel`, `exp`, `faction`, `npcflag`, `speed_walk`, `speed_run`, `speed_swim`, `speed_flight`, `detection_range`, `scale`, `rank`, `dmgschool`, `DamageModifier`, `BaseAttackTime`, `RangeAttackTime`, `BaseVariance`, `RangeVariance`, `unit_class`, `unit_flags`, `unit_flags2`, `dynamicflags`, `family`, `trainer_type`, `trainer_spell`, `trainer_class`, `trainer_race`, `type`, `type_flags`, `lootid`, `pickpocketloot`, `skinloot`, `PetSpellDataId`, `VehicleId`, `mingold`, `maxgold`, `AIName`, `MovementType`, `HoverHeight`, `HealthModifier`, `ManaModifier`, `ArmorModifier`, `ExperienceModifier`, `RacialLeader`, `movementId`, `RegenHealth`, `mechanic_immune_mask`, `spell_school_immune_mask`, `flags_extra`, `ScriptName`, `VerifiedBuild`) VALUES (200000, 0, 0, 0, 0, 0, 25635, 0, 0, 0, 'Amorrise Everbrook', 'Multivendor', 'Speak', 0, 80, 80, 2, 35, 129, 1, 1.14286, 1, 1, 1, 1.00, 1, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 'SmartAI', 0, 1, 1, 1, 1, 1, 0, 0, 1, 0, 0, 0, '', 12340);
``` 
Seems to work fine now so far.

**Comment by rtippenhauer** [comment]
I updated the table directly with no affect.  Rebooted the WorldServer, Still nothing, even seeing a name change, so I created I changed the entry to 200001 and updated the script, Reloaded both the World Server and Eluna with no success.  I also took your sql and added a new 200003 id and updated eluna and still not getting a second menu and my player:SendBroadcastMessage("Selected "..intid.." "..VENDOR_ID3) messages appear before the SendListInventory, but not after.
